ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,503 @@
+    satellite_latitudes = satellite_data['latitude'].values[time_mask]
+    hs_cal = satellite_data['hs_cal'].values[time_mask]
+    wsp = satellite_data['wsp'].values[time_mask]
+
+    time_dataarray = xr.DataArray(filtered_satellite_times_unix, dims=['time'], name='time', attrs={
+        'standard_name': 'time',
+        'units': 'seconds since 1970-01-01 00:00:00',
+        'calendar': 'standard',
+        'axis': 'T'
+    })
+
+    interpolated_ws_values, model_times_ws = prepare_and_interpolate(model_data_combined, 'wind_speed', satellite_latitudes, satellite_longitudes, filtered_satellite_times_unix, valid_times_unix)
+    interpolated_swh_values, model_times_swh = prepare_and_interpolate(model_data_combined, 'significant_wave_height', satellite_latitudes, satellite_longitudes, filtered_satellite_times_unix, valid_times_unix)
+
+    interpolated_dataset = xr.Dataset({
+        'time': time_dataarray,
+        'latitude': xr.DataArray(satellite_latitudes, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='latitude').assign_attrs(units='degree_north'),
+        'longitude': xr.DataArray(satellite_longitudes, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='longitude').assign_attrs(units='degree_east'),
+        'model_hs': xr.DataArray(interpolated_swh_values, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='model_hs').assign_attrs(units='m'),
+        'model_wnd': xr.DataArray(interpolated_ws_values, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='model_wnd').assign_attrs(units='m'),
+        'obs_hs': xr.DataArray(hs, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='obs_hs').assign_attrs(units='m'),
+        'obs_hs_cal': xr.DataArray(hs_cal, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='obs_hs_cal').assign_attrs(units='m'),
+        'obs_wnd': xr.DataArray(wsp, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='obs_wnd').assign_attrs(units='m/s'),
+        'obs_wnd_cal': xr.DataArray(wsp_cal, coords={'time': filtered_satellite_times_unix}, dims=['time'], name='obs_wnd_cal').assign_attrs(units='m/s'),
+        'fcst_hr': xr.DataArray(model_times_swh, dims=['time'], name='fcst_hr').assign_attrs(units='hours'),
+    })
+
+    interpolated_dataset.attrs['satellite_name'] = satellite_name
+    interpolated_dataset.attrs['model_name'] = model_name
+    initial_condition_time = extract_reference_times(filtered_grib_files[0])
+    initial_condition_time_unix = int(initial_condition_time) // 10**9
+    fcst_hr2 = (interpolated_dataset['fcst_hr'].values - initial_condition_time_unix) / 3600
+    interpolated_dataset['fcst_hr'] = xr.DataArray(fcst_hr2, dims=['time'], name='fcst_hr2').assign_attrs(description="Forecast hour relative to initial condition time", units='hours')
+    interpolated_dataset.attrs['initial_condition_time'] =f"{str(int(initial_condition_time_unix))} seconds since 1970-01-01 00:00:00"
```

Comment:
@Ghazal-Mohammadpour we'll want to be able to read this time in as a variable.  I'm concerned that having this basically as a string means that we will not be able to read this in from this netcdf file and use it.